rules/config_default.smk

```diff
@@ -33,333 +77,46 @@ class _SectionWrapper(object):
-            'alpha': __RECORD__(value=0.1, type=float),
-            'ignore_indels': __RECORD__(value=False, type=bool),
-            'posterior_threshold': __RECORD__(value=0.9, type=float),
-            'coverage': __RECORD__(value=0, type=int),
-            'shift': __RECORD__(value=3, type=int),
-            'keep_files': __RECORD__(value=False, type=bool),
-            'localscratch':  __RECORD__(value='', type=str),
-        }),
-        ('samtools_index', {
-            'mem': __RECORD__(value=2000, type=int),
-            'time': __RECORD__(value=20, type=int),
-            'conda': __RECORD__(value=f'{VPIPE_BASEDIR}/envs/lofreq.yaml', type=str),
-        }),
-        ('lofreq', {
-            'mem': __RECORD__(value=2000, type=int),
-            'time': __RECORD__(value=60, type=int),
-            'conda': __RECORD__(value='', type=str),
-
-            'consensus': __RECORD__(value=True, type=bool),
-            'extra': __RECORD__(value='', type=str),
-        }),
-        ('alignment_coverage', {
-            'mem': __RECORD__(value=1000, type=int),
-            'time': __RECORD__(value=60, type=int),
-            'conda': __RECORD__(value=f'{VPIPE_BASEDIR}/envs/smallgenomeutilities.yaml', type=str),
-
-            'coverage': __RECORD__(value=5, type=int),
-        }),
-        ('stats', {
-            'conda': __RECORD__(value=f'{VPIPE_BASEDIR}/envs/sam2bam.yaml', type=str),
-        }),
-        ('haploclique', {
-            'mem': __RECORD__(value=10000, type=int),
-            'time': __RECORD__(value=1435, type=int),
-            'conda': __RECORD__(value='', type=str),
-
-            'relax': __RECORD__(value=True, type=bool),
-            'no_singletons': __RECORD__(value=True, type=bool),
-            'no_prob0': __RECORD__(value=True, type=bool),
-            'clique_size_limit': __RECORD__(value=3, type=int),
-            'max_num_cliques': __RECORD__(value=10000, type=int),
-        }),
-        ('haploclique_visualization', {
-            'mem': __RECORD__(value=2000, type=int),
-            'time': __RECORD__(value=235, type=int),
-            'conda': __RECORD__(value=f'{VPIPE_BASEDIR}/envs/smallgenomeutilities.yaml', type=str),
-
-            'region_start': __RECORD__(value=0, type=int),
-            'region_end': __RECORD__(value=9719, type=int),
-            'msa': __RECORD__(value='', type=str),
-        }),
-        ('savage', {
-            'mem': __RECORD__(value=10000, type=int),
-            'time': __RECORD__(value=1435, type=int),
-            'threads': __RECORD__(value=0, type=int),
-            'conda': __RECORD__(value='', type=str),
-
-            'split': __RECORD__(value=20, type=int),
-        }),
-        ('web_visualization', {
-            'mem': __RECORD__(value=2000, type=int),
-            'time': __RECORD__(value=235, type=int),
-            'conda': __RECORD__(value=f'{VPIPE_BASEDIR}/envs/visualization.yaml', type=str),
-        }),
-    ])
-
-    def __init__(self):
+class VpipeConfig:
```

Comment:
Instead of using a custom configuration class, it might be a good idea to use Snakemake's built-in `configfile` [directive](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html#standard-configuration). One could then also make use of the built-in [validation capabilities](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html#validation).

Reply:
please see my update in the description of the pull request at the top of this page.

Reply:
I see, that's fair. In that case, @DrYak can comment further.

Reply:
I think I can fix the issue with the `threads` fallback. We just have to iterate once over the `config` dictionary after loading and evaluation.

Whats remaining is the decision if existing users are fine with rewriting config files.

Reply:
I'd think that the amount of new changes which we will introduce with the current set of open PRs warrants a major version increase. In that case breaking backwards compatibility is acceptable and we can expect end-users to rewrite their config files when they want to upgrade. Especially since the config files would need modifications anyway when switching to the virus-agnostic V-pipe version.
I am open to discuss this :-)

Reply:
I'd think that the amount of new changes which we will introduce with the current set of open PRs warrants a major version increase. In that case breaking backwards compatibility is acceptable and we can expect end-users to rewrite their config files when they want to upgrade. Especially since the config files would need modifications anyway when switching to the virus-agnostic V-pipe version.
I am open to discuss this :-)

Reply:
I fixed this now, see https://github.com/cbg-ethz/V-pipe/pull/93/commits/cf18ae4186b0f5b6d5d8d5625d4ded1aa3c3ef78, also mentioned below.

Please note that the nested configuration dictionaries are a bit cumbersome on the CLI, e.g. you have to use
```
$ ./vpipe -j 1 -p --verbose --debug --config general='{"aligner":"bwa"}'
```
(The part after `general=` is internally processed using `eval` and must be syntactically correct, regrettably the error message if this is not the case is cryptic and does not address the underlying issue)

Changes include:
- `vpipe.config` -> `vpipe_config.yaml`
- adaption of `init_project.sh` to copy `vpipe_config.yaml`
- adapting `rules/common.smk`
- removed `rules/config_default.smk`